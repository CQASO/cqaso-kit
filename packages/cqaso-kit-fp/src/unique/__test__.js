import unique from './unique.js';

test('should format list when a number list', () => {
    const formattedList = unique([1, 1, 2]);
    expect(formattedList).toEqual([1, 2]);
    expect(formattedList).toMatchSnapshot();
});

test('should format list when a string list', () => {
    const formattedList = unique(['a', 'b', 'a']);
    expect(formattedList).toEqual(['a', 'b']);
    expect(formattedList).toMatchSnapshot();
});

test('should format list when a undefined, null, NaN list', () => {
    const formattedList = unique(['a', 'a', undefined, undefined, null, null, NaN, NaN]);
    expect(formattedList).toEqual(['a', undefined, null, NaN]);
    expect(formattedList).toMatchSnapshot();
});
