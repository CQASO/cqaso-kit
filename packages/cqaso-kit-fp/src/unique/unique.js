
function unique(arr) {
    const n = []; //一个新的临时数组
    //遍历当前数组
    for(let i = 0; i < arr.length; i++){
      //如果当前数组的第i已经保存进了临时数组，那么跳过，
      //否则把当前项push到临时数组里面
      if (n.indexOf(arr[i]) == -1) n.push(arr[i]);
    }
    return n;
}

export default unique;
