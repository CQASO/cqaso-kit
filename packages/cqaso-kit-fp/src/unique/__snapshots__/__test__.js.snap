exports[`test should format list when a number list 1`] = `
Array [
  1,
  2,
]
`;

exports[`test should format list when a string list 1`] = `
Array [
  "a",
  "b",
]
`;

exports[`test should format list when a undefined, null, NaN list 1`] = `
Array [
  "a",
  undefined,
  null,
  NaN,
]
`;
